### ISSUE TITLE
Postgres driver: close() does not resolve.

### ISSUE THREAD

**Issue by michaelzangl** [typeorm/typeorm#7823]
## Issue Description

When calling connection.close() on a Postgres connection with many runners, the promise won't resolve. Node application won't quit due to some open sockets.

### Expected Behavior
Close should close all open sockets and then resolve

### Actual Behavior
Close won't resolve. No error

### Steps to Reproduce

```ts
connection.then(async c => {
        await c.close();
        console.log("CONNECTION CLOSED") // < This will never appear, every second connection to postgres server is still open
      });
```

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |      Linux    |
| Node.js version     | v16.3.0 | 
| Typescript version  | 4.0.3 | 
| TypeORM version     | 0.2.32 | 

### Relevant Database Driver(s)

- [ ] `aurora-data-api`
- [ ] `aurora-data-api-pg`
- [ ] `better-sqlite3`
- [ ] `cockroachdb`
- [ ] `cordova`
- [ ] `expo`
- [ ] `mongodb`
- [ ] `mysql`
- [ ] `nativescript`
- [ ] `oracle`
- [x] `postgres`
- [ ] `react-native`
- [ ] `sap`
- [ ] `sqlite`
- [ ] `sqlite-abstract`
- [ ] `sqljs`
- [ ] `sqlserver`

### Are you willing to resolve this issue by submitting a Pull Request?

- [x] Yes, I have the time, and I know how to start. (=> But I am not interested in a long discussion on how to refactor the whole module to do it nicely)
- [ ] Yes, I have the time, but I don't know how to start. I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

### Workaround: (1) patch the compiled file
In PostgresDriver.js:
```diff
                switch (_a.label) {
-                    case 0: return [4 /*yield*/, Promise.all(this.connectedQueryRunners.map(function (queryRunner) { return queryRunner.release(); }))];
+                    case 0: return [4 /*yield*/, Promise.all([...this.connectedQueryRunners].map(function (queryRunner) { return queryRunner.release(); }))];
                    case 1:
```

### Workaround: (2) fix the driver closePool Method
```ts
import {PostgresDriver} from "typeorm/driver/postgres/PostgresDriver";

        // @ts-ignore
        const oldClosePool = PostgresDriver.prototype.closePool;
        // @ts-ignore
        PostgresDriver.prototype.closePool = async function(pool) {
          await Promise.all([...this.connectedQueryRunners].map(queryRunner => queryRunner.release()));
          return await oldClosePool.bind(this)(pool);
        };
```

**Comment by imnotjames** [comment]
This was already fixed in https://github.com/typeorm/typeorm/pull/7596

**Comment by imnotjames** [comment]
Duplicate of #6958

**Comment by imnotjames** [comment]
Should be fixed already in 0.2.34
